writer.go

```diff
@@ -274,7 +287,15 @@ func (w *Writer) WriteMessages(ctx context.Context, msgs ...Message) error {
 			return io.ErrClosedPipe
 		}
 
-		for _, msg := range msgs {
+		for i, msg := range msgs {
+			if int(msg.message().size()) > w.config.BatchBytes {
+				err := MessageTooLargeError{
+					Message: msg,
+					Remaining: msgs[i:],
```

Comment:
```suggestion
					Remaining: msgs[i+1:],
```
I would expect to see `Remaining` contain the messages in the slice _after_ the message that's too large.  That way, I could feed them back into write.  WDYT?

Reply:
👍 I agree. I'll make that change.